lib/bundler/cli/outdated.rb

```diff
@@ -30,6 +30,13 @@ def run
         Bundler.definition(:gems => gems, :sources => sources)
       end
 
+      patch_level = [:major, :minor, :patch].select {|v| options.keys.include?(v.to_s) }
+      Bundler.definition.create_gem_version_promoter([]).tap do |gvp|
+        gvp.level = patch_level.first || :major
+        gvp.strict = options[:strict]
+      end
+      Bundler.load
```

Comment:
IMO, this is what we were trying to do in 1.12, and failed to do because resolving is complicated. :/ I would be okay with this change on the grounds that the previous flags were documented so it sounded like they cause the new resolver aware behavior. 👍